src/utils/strings.h

```diff
@@ -62,6 +73,10 @@ bool iequals(const char *lhs, const std::string &rhs, size_t n);
 // Decide whether the first n bytes of two memory areas are equal, even if one of them is NULL.
 bool mequals(const void *lhs, const void *rhs, size_t n);
 
+error_code pattern_match(const std::string &str,
```

Comment:
There is a similar implementation in `bool pegasus_server_impl::validate_filter(...)`, is it possible to unify to use it?

Reply:
OK, seems similar. I've added a `TODO` comment for this. And corner cases would also be checked later in case some errors occurred.